src/applications/vaos/components/cancel/CancelAppointmentFailedModal.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import Modal from '@department-of-veterans-affairs/formation-react/Modal';
+
+import FacilityAddress from '../FacilityAddress';
+
+export default function CancelAppointmentFailedModal({
+  appointment,
+  facility,
+  onClose,
+}) {
+  const clinicName =
+    appointment.clinicFriendlyName ||
+    appointment.vdsAppointments?.[0]?.clinic.name;
```

Comment:
Not a change request, but more of a question.  What's the best practice as far as optional chaining goes?  Should we add a `?` before each level when we use it?

Reply:
That's a good question. I've been trying to set it where I think there's a chance that an object might not exist. Like in this case, we might not have a `vdsAppointments` array and it might not have entries, but if it does, it looks like the `clinic` object is always there, so I didn't add one after before `name`.

I'm not sure this is the best rule. I don't want us to use it defensively, if we're just worried that something might go wrong, but it's sometimes hard to know when you may or may not need it.

Reply:
True.  I guess it's a similar predicament to try catch statements where if we over use optional chaining, we might miss out on potential errors.